### DISCUSSION TITLE
vs-mef with custom AssemblyLoadContext

### DISCUSSION THREAD

**Discussion by hymccord** [microsoft/vs-mef#299] [Show and tell]
v17 of vs-mef introduced a new overload to `ComposableCatalog.Create` that allows you to pass in a custom `Resolver`.

I wrote a some samples of using two different types of custom resolvers showing the capability of loading different major versions of `Newtonsoft.Json` in the same program.
You can see that here: https://github.com/InKahootz/vs-mef-alc

I thought it might be nice to show it off here and inspire others that have extensible apps where plugins come from other users.

Also, could we get an official nuget release (or at least of alpha) of v17? Currently I have to pack the vs-mef repo and locally source the nuget packages.

**Comment by AArnott** [comment]
Thanks for the ping. I've pushed a few more recent versions to nuget.org.
A sample for using this with an extensible app with many ALCs could be interesting. Where were you thinking of putting that sample?

**Reply by hymccord** [reply]
Much appreciated for the v17 push. Don't forget the Analyzers packages as well. Currently updating to the latest on NuGet causes package restore fail due to the Analyzers package still being on v16.

I can keep it in here for now since this is just a show and tell discussion and it's something I drafted. If it did go into the repo, I think a `samples/` folder at the root would be appropriate.

**Reply by AArnott** [reply]
Whoops. Usually an automated system pushes these packages as a set, but this time I did it manually so it led to my forgetting about the rest. I've pushed the additional packages.
